p2panda-core/src/prune.rs

```diff
@@ -82,4 +119,47 @@ mod tests {
         assert!(validate_prunable_backlink(None, &header, false).is_ok());
         assert!(validate_prunable_backlink(None, &header, true).is_ok());
     }
+
+    #[test]
+    fn prune_flag_encoding_is_short() {
+        let prune_flag = PruneFlag::default();
+        let bytes = encode_cbor(&prune_flag).unwrap();
+        assert_eq!(bytes.len(), 1);
+    }
+
+    #[test]
+    fn prune_flag_deref() {
+        let prune_flag = PruneFlag::default();
+        if *prune_flag {
+            panic!("should be false!");
+        }
+    }
+
+    #[test]
+    fn prune_flag_can_be_optional() {
+        #[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+        struct Extensions {
+            #[serde(
+                rename = "p",
+                skip_serializing_if = "PruneFlag::is_not_set",
```

Comment:
I don't mind the `is_not_set` method (I like how clear `PruneFlag::is_not_set` reads) but you could also use `std::ops::Not::not` instead.

Two options:

```rust
use std::ops::Not;

#[serde(skip_serializing_if = "Not::not")]

#[serde(skip_serializing_if = "<&bool>::not")]
```

Reply:
Oh, also very nice! Thank you!